### ISSUE TITLE
About Fred Dataset

### ISSUE THREAD

**Issue by 17011813** [DFrolova/ml-sk-project#1]
Hello!! 
Thanks to your code, I am studying well. :)
I have a question.
I saved Fred data as a txt file in **series_loader_final.ipynb**, but in **train_nbeats_1.ipynb** brought Fred data as csv file for training.
I'm confused about the Fred dataset type.

So, I tried to save Fred data as a txt file, and tried to train the csv Fred data in **train_nbeats_1.ipynb** but I've got some errors.

I wanna know more kind and specific explanations about your code!!

Thank you for reading my question.
Have a great day!

**Comment by DFrolova** [comment]
Hello!

Actually, **series_loader_final.ipynb** notebook was an intermediate step to download Fred. So I fixed google drive link to all downloaded data so that you can use it (you can find it in readme). There are csv files required for training inside.

All the best!

**Comment by 17011813** [comment]
Hello!! 

Thanks for your kind and fast answers :)

I have 3 more questions about your code implementation.

1. I tried your code with the datasets that you provided last night. I trained the model with Fred data(yearly) set, and I tested the model with M4 dataset(yearly), but I've got a different result than you specified.

Mean loss 108.26487264526996
TOTAL LOSS: 92.22305878728264
![n_beats_111](https://user-images.githubusercontent.com/48427281/134666203-63338d77-c65f-4901-ac2c-1d758b6968b9.png)

![n_beats_111](https://user-images.githubusercontent.com/48427281/134688689-e8b412b3-9148-49f8-bc06-b4c03b117050.png)

2. I wanna test other datasets, such as stock datasets. Have you experimented with simpler data other than the data specified in the paper? I want to try some univariate new datasets to train and test N-BEATS. How can I make it in your code?

3. Where is the meta-learning part of this train cord? I read and studied the paper[Meta-learning framework with applications to zero-shot time-series forecasting] and running this code, but I don't know how to meta-learn when using N-Beats in this code. Where is the meta-parameter in your code? I can't completely understand your code When I study your code with meta-learning.
I'm confused when studying meta-learning with your code.

Thank you for providing the perfect code!!!
Have a great day 👍

**Comment by DFrolova** [comment]
Hello!
1. Have you trained the whole ensemble of models? A single model performs much worse than an ensemble. Moreover, it can be bad because of random initialization of weights. The results are reported on an ensemble of 30 models (6 different horizons - forecast lengths - and 5 random initializations for each horizon).
2. You need to write function to load data of your dataset in the required format. It can be done similarly to functions get_<dataset>_data_testing.
3. As far as I understood, meta-learning is about training on one dataset and predicting on another.

**Comment by 17011813** [comment]
Hello there :)

Thanks for your kind and fast answers 👍 

I want to ask one more thing about your answer #3.

You said "meta-learning is about training on one dataset and predicting on another".

Then what's the difference between meta-learning and transfer learning?

Because as I know transfer learning also take the method that you mentioned in answer #3. 

Is the difference between meta-learning and transfer learning only using fine-tuning or not?

Thanks for reading my one more question!

Have a great day!

**Comment by DFrolova** [comment]
Hello!

As far as I understood it, yes, you are right. This approach does not require fine-tuning on a new dataset.

All the best!
